packages/react/src/Stack/Stack.tsx

```diff
@@ -68,34 +68,40 @@ type StackProps<As> = React.PropsWithChildren<{
 }>
 
 const StackBaseComponent = toggleSxComponent('div') as React.ComponentType<StackProps<React.ElementType>>
-function Stack<As extends ElementType>({
-  as,
-  children,
-  align = 'stretch',
-  direction = 'vertical',
-  gap,
-  justify = 'start',
-  padding = 'none',
-  wrap = 'nowrap',
-  className,
-  ...rest
-}: StackProps<As> & React.ComponentPropsWithoutRef<ElementType extends As ? As : 'div'>) {
-  return (
-    <StackBaseComponent
-      as={as}
-      {...rest}
-      className={clsx(className, classes.Stack)}
-      {...getResponsiveAttributes('gap', gap)}
-      {...getResponsiveAttributes('direction', direction)}
-      {...getResponsiveAttributes('align', align)}
-      {...getResponsiveAttributes('wrap', wrap)}
-      {...getResponsiveAttributes('justify', justify)}
-      {...getResponsiveAttributes('padding', padding)}
-    >
-      {children}
-    </StackBaseComponent>
-  )
-}
+const Stack = forwardRef(
+  <As extends ElementType>(
+    {
+      as,
+      children,
+      align = 'stretch',
+      direction = 'vertical',
+      gap,
+      justify = 'start',
+      padding = 'none',
+      wrap = 'nowrap',
+      className,
+      ...rest
+    }: StackProps<As> & React.ComponentPropsWithRef<ElementType extends As ? As : 'div'>,
+    forwardedRef: React.Ref<HTMLDivElement> | undefined,
```

Comment:
I think the type for this will get tricky because the `forwardRef` type should match whatever `As` is and I believe with the setup now it would always be `HTMLDivElement`.

The way this is typed also seems to be making the props inferred as `any` so they won't show up in autocomplete / won't fail if passed the wrong value (let me know if you're not seeing this locally though, could be something on my end)

```tsx
 <Stack
  as="span"
  ref={node => {
    // `node` is `HTMLDivElement` instead of `HTMLSpanElement`
  }}
  // I don't think this will fail TS now the way the types are inferred
  gap="nonsense"
/>
```

Reply:
Good catch! I didn't notice 😅. I **think** I fixed it now, lmk what you think!